### PR TITLE
Determined used and unused iam entities for AWS

### PR DESCRIPTION
**Pull request by ganeshjagdale9545**

### CODE REVIEW

File: cartography/intel/aws/iam.py [hunk 1]
```diff
@@ -925,6 +929,57 @@ def sync_user_access_keys(
     )
 
 
+def _set_used_state_tx(
+    tx: neo4j.Transaction, project_id: str, common_job_parameters: Dict, update_tag: int,
+) -> None:
+    ingest_role_used = """
+    MATCH (n:AWSRole)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}})
```
Comment (mpurusottamc) [inline cartography/intel/aws/iam.py:936]: same as other clouds, let's flip this query. start with CloudanixWorkspace->.... AWSRole

File: cartography/intel/aws/iam.py [hunk 2]
```diff
@@ -925,6 +929,57 @@ def sync_user_access_keys(
...
+    <-[:OWNER]-(:CloudanixWorkspace{id: {WORKSPACE_ID}})
+    WHERE (n)-[:TRUSTS_AWS_PRINCIPAL]->() AND n.lastupdated = {update_tag}
```
Comment (mpurusottamc) [inline cartography/intel/aws/iam.py:938]: please point me to the code where we determine TRUSTS_AWS_PRINCIPAL relationship

File: cartography/intel/aws/iam.py [hunk 3]
```diff
@@ -925,6 +929,57 @@ def sync_user_access_keys(
...
+    tx.run(
+        ingest_role_used,
+        WORKSPACE_ID=common_job_parameters['WORKSPACE_ID'],
+        update_tag=update_tag,
+        AWS_ID=project_id,
+        isUsed=True,
+    )
+
+    ingest_entity_used = """
+    MATCH (n)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}})
```
Comment (mpurusottamc) [inline cartography/intel/aws/iam.py:951]: same as above, let's flip this query.

File: cartography/intel/aws/iam.py [hunk 4]
```diff
@@ -925,6 +929,57 @@ def sync_user_access_keys(
...
+    tx.run(
+        ingest_entity_used,
+        WORKSPACE_ID=common_job_parameters['WORKSPACE_ID'],
+        update_tag=update_tag,
+        AWS_ID=project_id,
+        isUsed=True,
+    )
+
+    ingest_entity_unused = """
+    MATCH (n)<-[:RESOURCE]-(:AWSAccount{id: {AWS_ID}})
```
Comment (mpurusottamc) [inline cartography/intel/aws/iam.py:967]: same as above, let's flip this query.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
minor comments. needs query change.

File: cartography/intel/aws/iam.py (hunk 2 above)
Comment (ganeshjagdale9545) [reply cartography/intel/aws/iam.py:938]: Please can you explain little bit it? i can’t understand what you want to say

**Review by mpurusottamc — APPROVED** [review]
code looks good. lets' chat on Monday about the ASSUME_ROLE logic for all clouds.

**Pull request merged**